### ISSUE TITLE
ITV Episode Numbers

### ISSUE THREAD

**Issue by AtomicLaser** [Ascoware/get-iplayer-automator#141]
*What steps will reproduce the problem? Be specific, and provide as much detail as possible.*
Download Emmerdale/Coronation Street from ITV

*What is the expected output? What do you see instead?*
Downloads one file called ProgramName.S01e01.mp4 and then 'completes' the download for additional programes in the series with no extra files or errors It should download files sticking to the season/episode numbers

*What version of GiA are you using? On what version of OS X?*
GiA 1.12.0
Mac OS 10.13.3

*Are you using a web proxy? DNS proxy? VPN? VPS?*
no

*What program are you attempting to download (full title, series and episode)? **Provide only ONE program as an example.** *
as above

*Please provide any additional information below.*
Sample log:

Downloading Show 3/18:

ITV Download (ID=1a8694a8102): Downloading Emmerdale - Thu Mar 22
INFO: Preparing Request for Auth Info
INFO: Metadata processed.
ITV Download (ID=1a8694a8102): youtube-dl finished downloading
INFO: Downloading thumbnail
INFO: Thumbnail Download Completed
INFO: Beginning AtomicParsley Tagging.
INFO: AtomicParsley Tagging finished.
INFO: Converting Subtitles...
INFO: Converting to SubRip: /Users/myname/Movies/TV Shows/Emmerdale/Emmerdale.s01e01..en.vtt
ERROR: Conversion to SubRip failed: /Users/myname/Movies/TV Shows/Emmerdale/Emmerdale.s01e01..en.vtt
ffmpeg version 3.4.2-tessus Copyright (c) 2000-2018 the FFmpeg developers
  built with Apple LLVM version 9.0.0 (clang-900.0.39.2)
  configuration: --cc=/usr/bin/clang --prefix=/opt/ffmpeg --extra-version=tessus --enable-avisynth --enable-fontconfig --enable-gpl --enable-libass --enable-libbluray --enable-libfreetype --enable-libgsm --enable-libmodplug --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopus --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libtheora --enable-libvidstab --enable-libvo-amrwbenc --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libx264 --enable-libx265 --enable-libxavs --enable-libxvid --enable-libzmq --enable-libzvbi --enable-version3 --pkg-config-flags=--static --disable-ffplay
  libavutil      55. 78.100 / 55. 78.100
  libavcodec     57.107.100 / 57.107.100
  libavformat    57. 83.100 / 57. 83.100
  libavdevice    57. 10.100 / 57. 10.100
  libavfilter     6.107.100 /  6.107.100
  libswscale      4.  8.100 /  4.  8.100
  libswresample   2.  9.100 /  2.  9.100
  libpostproc    54.  7.100 / 54.  7.100
Input #0, webvtt, from '/Users/myname/Movies/TV Shows/Emmerdale/Emmerdale.s01e01..en.vtt':
  Duration: N/A, bitrate: N/A
    Stream #0:0: Subtitle: webvtt
File '/Users/myname/Movies/TV Shows/Emmerdale/Emmerdale.s01e01..en.srt' already exists. Overwrite ? [y/N] Not overwriting - exiting

**Comment by sleaux-meaux** [comment]
For now, don't use Kodi naming. If season/episodes numbers are not available, It appears that GiA doesn't construct a unique filename for each episode.

**Comment by wagnertoph** [comment]
Hi, I encounter the same issue. Not using Kodi naming doesn't quite solve it in all cases, because for example two episodes of Coronation Street broadcast on the same day will result in two names with the same file. The second then overwrites the first, so you end up with only one of the two files.

**Comment by AtomicLaser** [comment]
> @wagnertoph E.g #two episodes of Coronation Street broadcast on the same day will result in two names with the same file. The second then overwrites the first, so you end up with only one of the two files.

Ahh I hadn't thought of that being an issue - Currently testing how well not downloading the Kodi naming affects me as I download though GiA which then picks up on my plex server where I watch on the move or on a TV running amazon/plex so if its not able to pick the naming up from non kodi naming, then this isn't of use - However the comments by @wagnertoph above would be correct.

**Comment by skovatch** [comment]
This is tough because ITV doesn't send a numeric season and episode for these shows. You get    
```
  data-video-episode-id="1/0694/9410"
  data-video-programme-id="1/0694"
```
which isn't a simple numeric value. But episode-id is going to be unique.

**Comment by AtomicLaser** [comment]
I'm not sure how the kodi naming worked, but it worked well!

At the moment, turning off kodi naming downloads them as as "Emmerdale - Thu Mar 22.mp4" which does get picked up by plex, so at least it's sort of working - However there's still the issue where two episodes on the same day, one is downloaded and the other is downloaded and not saved because the file name already exists.

**Comment by sleaux-meaux** [comment]
As a workaround for same-day programme name clashes, edit the episode name in the download queue to add some sort of disambiguation before starting downloads. No idea how that might affect Plex scraping.

**Comment by jefro108** [comment]
With eg The Good Karma Hospital - Series 2 - Episode 2 https://www.itv.com/hub/the-good-karma-hospital/2a4663a0008 it gets downloaded at present with the Kodi naming setting on as `The Good Karma Hospital.s01e01.Series 2 - Episode 2.mp4` 

Since the data appears to be there to give it the name Series 2 - Episode 2, can that not be used to create the Kodi naming?

**Comment by sleaux-meaux** [comment]
youtube-dl can do the donkey work here. It would be a bit wasteful of bandwidth, but you could do an additional call to youtube-dl to return some JSON metadata from which you can pick out the elements available to create the output filename when downloading.

**Comment by skovatch** [comment]
I'm going to propose that for ITV shows, XBMC(Kodi) naming will always be used for the file name. I just have to make sure I parse out a valid season/episode combination or, if not available, the last part of the production ID which should be unique for each show. All show pages have a consistently-formatted JSON dictionary that I can parse for everything needed, so I should always be able to pick out the data I need.

If you import into iTunes the filename is irrelevant; it will display what's in the MP4 tags.

For example, Coronation Street sometimes airs multiple times a day, and has no series/episode information. In that case the file names would be:
```
Coronation Street.s01e9413.Mon Mar 26.mp4
Coronation Street.s01e9412.Mon Mar 26.mp4
```

A Touch Of Frost has series/episode number and episode names:
```
A Touch of Frost.s12e01.A Near Death Experience.mp4
```

James Martin's American Adventure has one series but numbered episodes:
```
James Martin's American Adventure.s01e19.Episode 19.mp4
```

I believe I have covered all the cases here, so unless I hear any strong objections I will release this as part of 1.12.2.

**Comment by wagnertoph** [comment]
That works for me, thank you so much.

**Comment by jefro108** [comment]
That's great for me! 😊

**Comment by bevanting** [comment]
I am running the latest version (1.12.2) and the naming is still not great. For instance, when downloading Hell's Kitchen, having two episode on the same day, it gave the same name. 
On Inspector Morse, instead of the episode number or name, it changed them to the day which I then had to change back..just a tad annoying.
However, I noticed that before pressing "Process ID", the names for each episode were correct. 
Can we not have an option to use the original names, set in Preferences?
